Modules/TOF/src/TOFMatchedTracks.cxx

```diff
@@ -337,24 +444,50 @@ void TOFMatchedTracks::monitorData(o2::framework::ProcessingContext& ctx)
         return true;
       }
       LOG(debug) << "DEN CONSTR: track with eta " << tpcTrack.getEta() << " and pT " << tpcTrack.getPt() << " ACCEPTED for denominator CONSTR, ITSTPC track";
-      this->mInTracksPt[trkType::CONSTR]->Fill(tpcTrack.getPt());
-      this->mInTracksEta[trkType::CONSTR]->Fill(tpcTrack.getEta());
-      this->mInTracks2DPtEta[trkType::CONSTR]->Fill(tpcTrack.getPt(), tpcTrack.getEta());
+      this->mInTracksPt[matchType::ITSTPC_ITSTPCTRD]->Fill(tpcTrack.getPt());
+      this->mInTracksEta[matchType::ITSTPC_ITSTPCTRD]->Fill(tpcTrack.getEta());
+      this->mInTracks2DPtEta[matchType::ITSTPC_ITSTPCTRD]->Fill(tpcTrack.getPt(), tpcTrack.getEta());
     }
+    // In case of ITS-TPC-TRD-TOF, TPC-TRD, ITS-TPC-TRD-TOF, TPC-TRD-TOF, it is always a TRD track
+    if constexpr (isTRDTrack<decltype(trk)>()) {
```

Comment:
Hello @shahor02 , 

Can you check that the selections in the lambda function are correct for the various track types? The resulting efficiencies look reasonable to me, but it is not so obvious to use the indices etc with the RecoContainer functionalities.

Reply:
Looks ok